### ISSUE TITLE
Make API-version passable when creating client.

### ISSUE THREAD

**Issue by felixlindstrom** [felixlindstrom/python-salesforce-api#12]

**Comment by mattschofield** [comment]
Hey @felixlindstrom just want to say thanks for maintaining this repo - there really is nothing else out there that's workable for the Salesforce REST APIs.

I notice in [this PR](https://github.com/felixlindstrom/python-salesforce-api/pull/13) you say that you've added support for specifying an API version. I think this is actually missing from the client initialisation. It looks to be everywhere except where you need it most, i.e.

```python
# This will fail as Salesforce does not expect an argument named "api_version"
client = Salesforce(access_token = 'SOME_TOKEN', domain = 'SOME_DOMAIN', api_version = '51.0')
```

I'm assuming that the fix would just be to add a field to the Client's initializer and pass this to Connection? Something like:

```python
class Client:
    def __init__(self,
        connection: Connection = None,
        domain: str = None,
        username: str = None,
        password: str = None,
        security_token: str = None,
        password_and_security_token: str = None,
        client_id: str = None, 
        client_secret: str = None,
        access_token: str = None,
        session: requests.Session = None,
        is_sandbox=False,

        # new parameter added
        api_version: str = None):

        self.connection = connection if connection else login.magic(
            domain=domain,
            username=username,
            password=password,
            security_token=security_token,
            password_and_security_token=password_and_security_token,
            client_id=client_id,
            client_secret=client_secret,
            access_token=access_token,
            session=misc_utils.get_session(session),
            is_sandbox=is_sandbox,

            # new parameter passed to Connection
            api_version=api_version
        )
        self._setup_services()
```

I may be missing something here but let me know if you think my diagnosis is correct!

**Comment by felixlindstrom** [comment]
Hello @mattschofield!

Thank you for the comment!

I think you are completely correct!

Do you want to create a PR for this, and contribute to the project? If not, I'd be happy to apply the change and create a new build for pypi!

**Comment by felixlindstrom** [comment]
@mattschofield This is now, through your merge, available in the published version 0.1.44! Thank you for the help!
